### PR TITLE
Add Clang Formatting

### PR DESCRIPTION
**Pull request by mpowelson**
Adds Clang formatting to CI build as well as a script to run it.

Builds on #10

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+jobs:
+  include:
+    - os: linux
+      dist: bionic
+      language: cpp
+      env:
+      - ROS_DISTRO="melodic"
+      - CLANG_FORMAT_CHECK=file
+      - CLANG_FORMAT_VERSION=8
+      - DOCKER_IMAGE=lharmstrong/tesseract:melodic
```
Comment (marip8) [inline .travis.yml]: Why are we using a tesseract docker image here?

Comment (mpowelson) [reply .travis.yml]: It probably shouldn't. I can remove this.

Comment (marip8) [reply .travis.yml]: Maybe we can try the regular ROS Melodic image like you mentioned [here](https://github.com/swri-robotics/Toolpath-Offline-Planning/pull/10#issuecomment-575207519)

**Review by marip8 — APPROVED** [review]

**Comment by marip8** [comment]
Addresses #4

**Pull request merged**